include/mav/MessageSet.h

```diff
@@ -70,6 +83,53 @@ namespace mav {
                 _document(std::move(document)),
                 _root_xml_folder_path(root_xml_folder_path) {}
 
+        
+        static inline uint64_t _strict_stoul(const std::string &str, int base=10) {
+            if (str.empty()) {
+                throw ParseError("Enum value is empty string");
+            }
+
```

Comment:
At the moment, you could input a negative value, which gets convered to an unsigned. You could check here as well that the string only contains numbers/hex values?